Mathlib/MeasureTheory/Integral/Layercake.lean

```diff
@@ -377,3 +381,95 @@ theorem lintegral_rpow_eq_lintegral_meas_lt_mul (μ : Measure α) [SigmaFinite 
 #align lintegral_rpow_eq_lintegral_meas_lt_mul lintegral_rpow_eq_lintegral_meas_lt_mul
 
 end LayercakeLT
+
+section LayercakeIntegral
+
+variable {α : Type*} [MeasurableSpace α] {μ : Measure α} [SigmaFinite μ]
+
+/-- If `f` is integrable, then for any `c > 0` the set `{x | ‖f x‖ ≥ c}` has finite measure. -/
+lemma Integrable.measure_const_le_norm_lt_top {E : Type*} [NormedAddCommGroup E] [MeasurableSpace E]
+    [BorelSpace E] {f : α → E} (f_intble : Integrable f μ) {c : ℝ} (c_pos : 0 < c) :
+    μ {a : α | c ≤ ‖f a‖} < ∞ := by
+  have norm_f_aemble : AEMeasurable (fun a ↦ ENNReal.ofReal ‖f a‖) μ :=
+    (ENNReal.measurable_ofReal.comp measurable_norm).comp_aemeasurable f_intble.1.aemeasurable
+  have markov := mul_meas_ge_le_lintegral₀ (μ := μ) norm_f_aemble (ENNReal.ofReal c)
+  have obs : ∫⁻ (a : α), ENNReal.ofReal ‖f a‖ ∂μ = ∫⁻ (a : α), ‖f a‖₊ ∂μ :=
+    lintegral_congr (fun x ↦ ofReal_norm_eq_coe_nnnorm (f x))
+  simp_rw [ENNReal.ofReal_le_ofReal_iff (norm_nonneg _), obs] at markov
+  have c_inv_ne_top : (ENNReal.ofReal c)⁻¹ ≠ ∞ :=
+    ENNReal.inv_ne_top.mpr (ENNReal.ofReal_pos.mpr c_pos).ne.symm
+  simpa [← mul_assoc,
+         ENNReal.inv_mul_cancel (ENNReal.ofReal_pos.mpr c_pos).ne.symm ENNReal.ofReal_ne_top]
+    using ENNReal.mul_lt_top c_inv_ne_top (lt_of_le_of_lt markov f_intble.2).ne
+
+/-- If `f` is integrable, then for any `c > 0` the set `{x | ‖f x‖ > c}` has finite measure. -/
+lemma Integrable.measure_const_lt_norm_lt_top
+    {E : Type*} [NormedAddCommGroup E] [MeasurableSpace E] [BorelSpace E]
+    {f : α → E} (f_intble : Integrable f μ) {c : ℝ} (c_pos : 0 < c) :
+    μ {a : α | c < ‖f a‖} < ∞ :=
+  lt_of_le_of_lt (measure_mono (fun _ h ↦ (Set.mem_setOf_eq ▸ h).le))
+    (Integrable.measure_const_le_norm_lt_top f_intble c_pos)
+
+variable {f : α → ℝ} (f_intble : Integrable f μ)
```

Comment:
could you move the assumptions `f_intble` to all the lemmas that use it? Otherwise, the statements are harder to understand for someone just browsing through the file.